report_thesis/src/sections/proposed_approach/optimization_framework.tex

```diff
@@ -42,152 +40,125 @@ \subsubsection{The Framework}
-Similar to the sampler, there exist several options for the choice of pruner.
-In our case we chose to use the Hyperband pruner, due to its balanced approach when pruning.
-Initially, the pruner will consider more configurations, but allow them to run for a shorter time.
-Through several iterations, it will gradually reduce the number of configurations being considered, but allow them to run for a longer time.
-This provides a higher likelihood that promising configurations are not pruned too early, while still reducing the computational costs.\cite{optuna_2019}
+Guiding the optimization process is the \nameref{alg:combined_objective} function, which evaluates the performance of each trial, seeking to either minimize or maximize some specified metrics.
+In our case we are seeking to minimize the \texttt{rmse\_cv} and \texttt{std\_dev\_cv}, as mentioned previously.
 
-Guiding both the sampler and the pruner is the objective function, which evaluates the performance of each trial, seeking to either minimize or maximize some specified metrics.
-In our case we are seeking to minimize the \texttt{rmse\_cv} and \texttt{std\_dev\_cv}, defined previously.
-This can be seen in line~\ref{step:initialize_objective}.
-
-The role of the objective function is to provide the metric data that the sampler and pruner need to make informed decisions.
-As we step through each oxide and model in Lines~\ref{step:oxide_loop} to~\ref{step:model_loop}, we call the $objective$ function with the $Optimize\_Combined\_Objective$ function, seen in Line~\ref{step:optimize_combined_objective}.
-The $Optimize\_Combined\_Objective$ is returning the metrics for each trial, which the objective function then uses to guide the optimization process.
+The role of the \nameref{alg:combined_objective} function is to provide the metric data to an $optimize$ function, seen in Line~\ref{step:optimize_objective}.
+As we step through each oxide and model in Lines~\ref{step:oxide_loop} to~\ref{step:model_loop}, we call the $optimize$ function with the number of trials to run and the \nameref{alg:combined_objective} function.
+The number of trials is an important parameter, as it regulates how much time the framework has to optimize a given model.
+In an ideal scenario, this number would be very high to ensure that the optimization process has identified the best possible configuration.
+However, depending on the number of models in consideration the number of trials can quickly become computationally prohibitive.
+The $optimize$ function uses the number of trials and the objective of minimizing the \texttt{rmse\_cv} and \texttt{std\_dev\_cv} to manage the optimization process and mediating the metrics returned by the \nameref{alg:combined_objective} function to the sampler. 
 
 This leads us to the \nameref{alg:combined_objective} function which, as previously mentioned, contains the optimization logic.
```

Comment:
It doesn't contain the optimization logic
```suggestion
This leads us to the \nameref{alg:combined_objective} function which, as previously mentioned, returns the evaluation metrics for the given trial.
```

but this is pretty weird because you literally introduce it above, and then say 'this leads us to' (it)?

Reply:
76, 77, and 79 are confusing? Seems like remnants of old text